### PR TITLE
Fix unmarshall errors, add condition for `nil reference`

### PR DESCRIPTION
**Pull request by illia-li**
Added for `smallint`, `int`, `bigint`, `counter` `cql types`.
Additional fixed `errors text` for `tinyint` `cql type`

### CODE REVIEW

File: marshal/cqlint/unmarshal_utils.go [hunk 1]
```diff
@@ -323,7 +399,7 @@ func decReflectInt8(p []byte, v reflect.Value) error {
 	case 4:
 		val := decInt32(p)
 		if val > math.MaxInt8 || val < math.MinInt8 {
-			return fmt.Errorf("failed to unmarshal int: to unmarshal into int8, the data should be in the int8 range")
+			return fmt.Errorf("failed to unmarshal int: to unmarshal into custom int8, the data should be in the int8 range")
```
Comment (dkropachev) [inline marshal/cqlint/unmarshal_utils.go:402]:
```suggestion
			return fmt.Errorf("failed to unmarshal int: to unmarshal into %T, the data should be in the int8 range", v.Interface())
```

**Review by dkropachev — CHANGES_REQUESTED** [review]

File: marshal/bigint/unmarshal.go [hunk 2]
```diff
@@ -71,7 +71,7 @@ func Unmarshal(data []byte, value interface{}) error {
 		rv := reflect.ValueOf(value)
 		rt := rv.Type()
 		if rt.Kind() != reflect.Ptr {
-			return fmt.Errorf("failed to unmarshal bigint: unsupported value type (%T)(%#[1]v)", value)
+			return fmt.Errorf("failed to unmarshal bigint: unsupported value type %#v", value)
```
Comment (dkropachev) [inline marshal/bigint/unmarshal.go:74]: Why did you change it ? now you don't see type in the error message

File: marshal/bigint/unmarshal_utils.go [hunk 3]
```diff
@@ -406,7 +482,7 @@ func DecBigIntR(p []byte, v **big.Int) error {
 
 func DecReflect(p []byte, v reflect.Value) error {
 	if v.IsNil() {
-		return fmt.Errorf("failed to unmarshal bigint: can not unmarshal into nil reference (%T)(%#[1]v)", v.Interface())
+		return fmt.Errorf("failed to unmarshal bigint: can not unmarshal into nil reference %#v)", v.Interface())
```
Comment (dkropachev) [inline marshal/bigint/unmarshal_utils.go:485]: Same question here

**Review by dkropachev — CHANGES_REQUESTED** [review]

File: marshal/bigint/unmarshal.go (hunk 2 above)
Comment (illia-li) [reply marshal/bigint/unmarshal.go:74]: `%#v` - include type

File: marshal/bigint/unmarshal_utils.go (hunk 3 above)
Comment (illia-li) [reply marshal/bigint/unmarshal_utils.go:485]: `%v` - type excluded 
`%#v` - type included

Comment (dkropachev) [reply marshal/bigint/unmarshal_utils.go:485]:
```
type Custom string

func Test1(t *testing.T) {
	t.Log(fmt.Errorf("%#v", Custom("123")))
}
```

Produces:
```
=== RUN   Test1
    scylla_shard_aware_port_common_test.go:21: "123"
--- PASS: Test1 (0.00s)
PASS 
```

Comment (illia-li) [reply marshal/bigint/unmarshal_utils.go:485]: Yeap, but:
```
func TestSimplestTest2(t *testing.T) {
	type cInt int
	var v interface{}
	var vv *cInt
	v = vv
	fmt.Println(errNilReference(v))
}

func errNilReference(v interface{}) error {
	return fmt.Errorf("failed to unmarshal varint: can not unmarshal into nil reference %#v", v)
}
```
Produced:
```
=== RUN   TestSimplestTest2
failed to unmarshal varint: can not unmarshal into nil reference (*main.cInt)(nil)
--- PASS: TestSimplestTest2 (0.00s)
```

Comment (illia-li) [reply marshal/bigint/unmarshal_utils.go:485]: Possible your example have build-in `Stringer` interface

Comment (illia-li) [reply marshal/bigint/unmarshal_utils.go]: Changed to `(%T)(%[1]v)`

**Comment by dkropachev** [comment]
@illia-li , could you please rebase

**Comment by illia-li** [comment]
> @illia-li , could you please rebase

Done

**Review by dkropachev — APPROVED** [review]
